### ISSUE TITLE
Remove ignored folders?

### ISSUE THREAD

**Issue by radzio** [keybase/client#5599]
Hello!

Is it possible to remove ignored folders using gui / cmd? I don't see them in Windows Explorer and I don't want to see them in the GUI too.

![image](https://cloud.githubusercontent.com/assets/469111/22415784/45bc9808-e6ca-11e6-982f-635f93b7efe8.png)

**Comment by cjb** [comment]
No -- it's letting you know that they are ignored so that you aren't later confused about why they aren't in Explorer.

CC @malgorithms

**Comment by radzio** [comment]
Any chances to add support for this? It might be a problem if someone follows more peoples but does not want to have folders with them.

@cjb  @malgorithms

**Comment by cjb** [comment]
@radzio I don't think so. If you could remove them from that list, how would you ever unignore them if you later change your mind?

**Comment by radzio** [comment]
@cjb  maybe add something here:

![image](https://cloud.githubusercontent.com/assets/469111/22591224/ac249068-ea13-11e6-9dd8-e0f4311e2309.png)

It's strange but keybase does not show ignored folders for all people that I follow ( I don't have any folders for them). It seems that keybase has started creating these folders recently for new people I've started to follow fyi.

**Comment by strib** [comment]
@radzio: in a release a few weeks ago, we had a bug that caused folders to be created with people you followed.  This was unintentional; folders should only be created when putting data in there.  It's fixed now; sorry for the inconvenience.

**Comment by radzio** [comment]
@strib  thanks for information. So this explains why these folders were created. Now I can't remove them from app :( ( I've removed them using rm -Rf  from keybase drive) but I can't get rid of them from keybase app :(. Any suggestions?

**Comment by strib** [comment]
I thought the folders were already in your "Ignored folders" list?  As @cjb explained above, we purposely don't let you remove them from the list.

If they're in your main folders list, you can remove them by selecting them in the GUI and choosing "Remove", or use `rmdir` from the terminal.

**Comment by radzio** [comment]
@strib  nope. Keybase created these folders automatically when I've started to follow these users. I've removed these folders using rmdir ( I didn't want them anyway ), and now I get stuck with unwanted ignored folders - that's why I want to remove them :/

**Comment by strib** [comment]
Ok got it, thanks.  There's no way to remove ignored folders, as stated above.  You can just keep your ignored list collapsed if you don't want to see them on a daily basis.

**Comment by radzio** [comment]
Thanks for patience and help ;-). I still think that users should be able to remove ignored folders too but I get your point. Cheers!

**Comment by ihackalot** [comment]
I second @radzio there should be a way to get rid of _junk_. My actual understanding is that those folders have to live forever, and that's the reason why you decide to keep an _ignored folders_ list. I'm not comfortable with stuff living forever 🤔

**Comment by radzio** [comment]
@ihackalot  👍

**Comment by johanbove** [comment]
Kind on topic I hope; hope does one un-ignore folders in the desktop client? ❓ 
Using Version 1.0.18-20170208165022+dc66461

**Comment by cjb** [comment]
@johanbove Huh.  But your own private folder can't be *ignored* either, right?  So it doesn't seem like a problem that it can't be unignored.  Maybe give more details about what you're trying to do?

**Comment by johanbove** [comment]
Thanks for the reply @cjb ; I guess I am confused about my own private folder being ignored. As a user it seems weird how my own folder is being "ignored". I probably don't understand the concept of "ignored" folders correctly. 😉 

![keybase-private-ignored-folder-v1](https://cloud.githubusercontent.com/assets/922765/23035042/2dcc906a-f47e-11e6-9992-3a7ecab59d7c.png)

(screenshot is from Windows 10 where my own private folder also looks ignored)

**Comment by cjb** [comment]
Weird!  I think the bug is that you were able to ignore that folder, rather than that you can't unignore it.  @strib, know of a way to do an unignore without the GUI?

**Comment by johanbove** [comment]
@strib The file was created without issues but the folder is still being shown as "ignored" in the GUI. I can however open the folder buy clicking the "open" button in the GUI. The files inside the private/johanbove folder can be edited and read normally.

**Comment by strib** [comment]
Ok, thanks for trying.  Weird.  Does the folder _also_ show in your normal list of folders, or only in the ignore list?

cc: @maxtaco @oconnor663

**Comment by johanbove** [comment]
@strib The folder only displays in the Keybase GUI in the "ignored folders" section. Un-ignoring and ignoring the other user's and shared folders works fine otherwise. 
Perhaps an exception needs to be made in the GUI for the user's personal private folder so that it cannot be "ignored" or is shown in a different section or manner?

**Comment by johanbove** [comment]
Here is my log file in case it could be helpful: https://github.com/keybase/client/issues/new?body=[write%20something%20useful%20and%20descriptive%20here]%0A%0Amy%20log%20id:%20719a9eb1bb1c5b04da30561c

**Comment by oconnor663** [comment]
@strib @cjb we have a `keybase favorite add/remove` command in the CLI for manipulating these things directly, but it's only enabled for devel builds. Do you think we should expose it in prod?

I assume that @johanbove got into this state with `rmdir` on his personal folder. When I try that with my test user, it repro's this, though the folder still shows up when I `ls /keybase/private`.

**Comment by strib** [comment]
@oconnor663: ah, yeah, it's possible KBFS doesn't call the favorite add when it's one of your personal folders, so an explicit command for it would help.  Seems reasonable to have that as an undocumented command in the prod builds, to help with cases like this.

In this case, can we just fix up his ignored list manually in the server?

**Comment by johanbove** [comment]
Thanks for the follow up on this. I actually think it's not just a thing on my account only since in @radzio 's original screenshot it also seems that his personal "private" folder is ignored.

**Comment by oconnor663** [comment]
@johanbove I just fixed this for you in our DB. Can you confirm that it's fixed?

**Comment by johanbove** [comment]
Yep. That fixed it alright. Thanks 👍

**Comment by strib** [comment]
> Can you fix my public folder?

@oconnor663?

> And was me getting a bunch of public folders from reading public messages by design?
> I now use peak as part of my read and haven't had any more public folders show up, but I don't know if that is because they are all ignored.

That was a bug that's been fixed in recent releases.  Sorry!

**Comment by oconnor663** [comment]
@strib @coldwaterq should be fixed in the DB now.

**Comment by coldwaterq** [comment]
Thank you

**Comment by radzio** [comment]
@oconnor663  can you please remove all my private ignored folders (leave only my /radzio folder)  on the server side? It's really annoying that I can't do this by myself and now I see these folders on every device :(

**Comment by strib** [comment]
Hi @radzio -- what specifically bothers you?  Just seeing them listed under "Ignored folders" in the GUI?  If so, you can click on the little triangle icon next to "Ignored folders" and the whole list will collapse, and you won't have to look at them anymore.

**Comment by radzio** [comment]
@strib because I didn't create these folders ( there were created without my knowledge-> https://github.com/keybase/client/issues/5599#issuecomment-277295782), I will never use them. I just want to have more control. If I remove something = it should be removed, if I ignore something it should be ignored. If you don't support removing folders once and for all at least please delete folders which weren't created by me :(

**Comment by coldwaterq** [comment]
For what it's worth I feel the same way about the ones that I have. Mine
were cause by reading public chats of people, so I assume the folders show
up because I have a file in each of their folders saying I read those chats.

**Comment by strib** [comment]
Ok, thanks for the explanation. At the moment our backend data model doesn't allow for folders to be something less than ignored -- if we remove them from the ignored list, they will just show up as new folders to you, and probably be more annoying.

We'll consider adding a complete removal for some point in the future, but at the moment we have higher-priority work ahead of it.  Sorry for the inconvenience.

**Comment by strib** [comment]
@coldwaterq: those chat folders showed up due to a bug in our initial version of chat, they weren't supposed to be added.  Sorry for the inconvenience.
